cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -768,6 +792,11 @@ fromConsensusQueryResultShelleyBased _ QueryCurrentEpochState q' r' =
     Consensus.GetCBOR Consensus.DebugEpochState -> SerialisedCurrentEpochState r'
     _                                           -> fromConsensusQueryResultMismatch
 
+fromConsensusQueryResultShelleyBased _ QueryPoolState{} q' r' =
+  case q' of
+    Consensus.GetCBOR Consensus.GetPoolState {} -> SerialisedPoolState r'
+    _                                           -> error "moomoo"
```

Comment:
It does.  Left over debugging.  Fixing now.

Reply:
Something that needs improvement here is that when `fromConsensusQueryResultMismatch` fails, I don't know which branch it is that failed because `fromConsensusQueryResultMismatch` is called from so many places.  There might be some opportunity to use `HasCallstack` to get a better experience here.

Reply:
> Something that needs improvement here is that when fromConsensusQueryResultMismatch fails

Can you paste the failure here?

Reply:
It just says `fromConsensusQueryResult: internal query mismatch`